src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+    "Then, to use that model to predict classes for the dataset '" +
+    PRINT_DATASET("test") + "', storing the output predictions in '" +
+    PRINT_DATASET("predictions") + "', the following command may be used: "
+    "\n\n" +
+    PRINT_CALL("linear_svm", "input_model", "lsvm_model", "test", "test",
+        "predictions", "predictions"),
+    SEE_ALSO("@random_forest", "#random_forest"),
+    SEE_ALSO("@logistic_regression", "#logistic_regression"),
+    SEE_ALSO("LinearSVM on Wikipedia",
+        "https://en.wikipedia.org/wiki/Support-vector_machine"),
+    SEE_ALSO("mlpack::svm::LinearSVM C++ class documentation",
+        "@doxygen/classmlpack_1_1svm_1_1LinearSVM.html"));
+
+// Training parameters.
+PARAM_MATRIX_IN("training", "A matrix containing the training set (the matrix "
+    "of predictors, X).", "t");
+PARAM_UROW_IN("labels", "A matrix containing labels (0 or 1) for the points "
+    "in the training set (y).", "l");
+
+// Optimizer parameters.
+PARAM_DOUBLE_IN("lambda", "L2-regularization parameter for training.", "L",
+    0.0001);
+PARAM_DOUBLE_IN("delta", "Margin of difference between correct class and other "
+    "classes.", "d", 1.0);
+PARAM_INT_IN("number_of_classes", "Number of classes for classification; if "
+    "unspecified (or 0), the number of classes found in the labels will be "
+    "used.", "c", 0);
+PARAM_FLAG("no_intercept", "Do not add the intercept term to the model.", "N");
+PARAM_STRING_IN("optimizer", "Optimizer to use for training ('lbfgs' or "
+    "'psgd').", "O", "lbfgs");
+PARAM_DOUBLE_IN("tolerance", "Convergence tolerance for optimizer.", "e",
+    1e-10);
+PARAM_INT_IN("max_iterations", "Maximum iterations for optimizer (0 indicates "
+    "no limit).", "n", 10000);
+#ifdef HAS_OPENMP
+PARAM_DOUBLE_IN("step_size", "Step size for ParallelSGD optimizer.", "s", 0.01);
+PARAM_FLAG("shuffle", "Don't shuffle the order in which data points are "
+    "visited for ParallelSGD.", "S");
+#endif
+// Model loading/saving.
+PARAM_MODEL_IN(LinearSVM<>, "input_model", "Existing model "
+    "(parameters).", "m");
+PARAM_MODEL_OUT(LinearSVM<>, "output_model", "Output for trained "
+    "linear svm model.", "M");
+
+// Testing.
+PARAM_MATRIX_IN("test", "Matrix containing test dataset.", "T");
+PARAM_UROW_IN("test_labels", "Matrix containing test labels.", "A");
+PARAM_UROW_OUT("predictions", "If test data is specified, this matrix is where "
+    "the predictions for the test set will be saved.", "P");
+PARAM_MATRIX_OUT("score", "If test data is specified, this "
+    "matrix is where the class score for the test set will be saved.",
+    "p");
+
+size_t NumberOfClasses(const size_t numClasses,
+                       const arma::Row<size_t>& labels);
+
+static void mlpackMain()
+{
+  // Collect command-line options.
+  const double lambda = CLI::GetParam<double>("lambda");
+  const double delta = CLI::GetParam<double>("delta");
+  const bool intercept = CLI::HasParam("no_intercept") ? true : false;
```

Comment:
No need for the `? true : false` here; `HasParam()` already returns a `bool`.

Reply:
Done